Header header
LineSegment[] line_segments
